tx.go

```diff
@@ -195,14 +208,16 @@ func (tx *Tx) Commit() error {
 		// tx.rollback()
 		// return errors.New(lackOfDiskSpace)
 		if err := tx.db.grow(int(tx.meta.Pgid()+1) * tx.db.pageSize); err != nil {
+			lg.Errorf("growing db size failed, pgid: %d, pagesize: %d, error: %v", tx.meta.Pgid(), tx.db.pageSize, err)
 			tx.rollback()
 			return err
 		}
 	}
 
 	// Write dirty pages to disk.
 	startTime = time.Now()
-	if err := tx.write(); err != nil {
+	if err = tx.write(); err != nil {
+		lg.Errorf("writing data failed: %v", err)
```

Comment:
I think it's duplicate in `write` and `writeMeta`. Please consider to remove this.

Reply:
Yes, since `tx.Commit` is the most important top-level (public) method in bbolt. It would be a little better to print an error message for each error in top-level even the underneath methods already printed similar error messages.

In general, I agree that we should print as much useful log as possible and also avoid duplication as much as we can. We can continue to enhance the log messages on top of this PR.